core/src/main/java/com/oath/oak/EntrySet.java

```diff
@@ -930,30 +692,30 @@ boolean deleteValueFinish(LookUp lookUp) {
      * therefore the key is written to off-heap and associated with the entry simultaneously.
      * The value of the new entry is set to NULL: <INVALID_VALUE_REFERENCE, INVALID_VERSION></>
      **/
-    LookUp allocateEntry(K key) {
+    void allocateEntry(ThreadContext ctx, K key) {
+        ctx.invalidate();
+
         int ei = nextFreeIndex.getAndIncrement();
         if (ei > entriesCapacity) {
-            return null;
+            return;
         }
         numOfEntries.getAndIncrement();
         int intIdx = entryIdx2intIdx(ei);
         // key and value must be set before returned
         // setting the value reference to <INVALID_VALUE_REFERENCE, INVALID_VERSION>
         // TODO: should we do the following setting? Because it is to set zero on zero...
-        setEntryFieldLong(intIdx, OFFSET.VALUE_REFERENCE, INVALID_VALUE_REFERENCE);
-        setEntryFieldInt(intIdx, OFFSET.VALUE_VERSION, INVALID_VERSION);
+        setEntryFieldLongVolatile(intIdx, OFFSET.VALUE_REFERENCE, ReferenceCodec.INVALID_REFERENCE);
+        setEntryFieldInt(intIdx, OFFSET.VALUE_VERSION, Slice.INVALID_VERSION);
 
-        writeKey(key, ei);
-        return new
-            LookUp(null,INVALID_VALUE_REFERENCE,ei,INVALID_VERSION,getKeyReference(ei), false);
+        writeKey(ctx, key, ei);
     }
 
-    boolean isEntryDeleted(int ei) {
-        int[] valueVersion = new int[1];
-        long valueReference = getValueReferenceAndVersion(ei, valueVersion);
-        return (valueReference == INVALID_VALUE_REFERENCE) ||
-            valOffHeapOperator.isValueDeleted(
-                buildValueSlice(valueReference, valueVersion[0], memoryManager), valueVersion[0]) != FALSE;
+    boolean isEntryDeleted(ValueBuffer value, int ei) {
+        boolean isValid = readValue(value, ei);
```

Comment:
readValue() also includes reading the buffer (allocator.readByteBuffer(s)) which is not necessary here... I also see that isEntryDeleted is used only in rebalance scenario, where reading the buffer is indeed not needed. Can we eliminate it?

Reply:
_Possible solution:_ add a helper method: `readValueWithoutBuffer()` that will be called by `isRntryDeleted()` and by the "regular" `readValue()` method.

Reply:
**Correctiontion**: `isEntryDeleted()` calls `valOffHeapOperator.isValueDeleted(value)` which reads the off-heap version from the buffer. So it needs the buffer.
Nothing needs to be changed then.